src/Sulu/Bundle/MediaBundle/Resources/js/containers/Form/fields/MediaSelection.js

```diff
@@ -16,12 +18,9 @@ export default class MediaSelection extends React.Component<FieldTypeProps<Value
 
     render() {
         const {formInspector, disabled, value} = this.props;
-
-        if (!formInspector || !formInspector.locale) {
-            throw new Error('The media selection needs a locale to work properly');
-        }
-
-        const {locale} = formInspector;
+        const locale = formInspector && formInspector.locale
+            ? formInspector.locale
+            : computed(() => userStore.contentLocale);
```

Comment:
Why the `computed`? Same question for all the other components

Reply:
the media-bundle components expect an observable/boxed string as locale. this is because they need to update the language of the displayed medias when the locale of the form is changed.

the `computed` function returns a boxed observable based on the return value of the given function. furthermore, the returned observable will update itself when an observable that is used in the function (in this case the `contentLocale` of the `userStore`) changes.

Reply:
Unfortunatelly, the object returned by the `computed` function does not provide an `intercept` method and therefore does not work inside a `DatagridStore` 🙁 Therefore i changed the components to use `observable.box`.